fastparquet/writer.py

```diff
@@ -137,7 +140,10 @@ def convert(data, se):
         out = data.values
     elif dtype == "O":
         if converted_type == parquet_thrift.ConvertedType.UTF8:
-            out = array_encode_utf8(data)
+            if PY2:
+                out = np.array([x.encode('utf8') for x in data], dtype="O")
```

Comment:
No such luck.  array_encode_utf8 is not friendly to Python2 at all.  The performance issue is a known limitation, at least for now.